File: API/domain/Entities/crm/Company.cs
using System;
using System.Collections.Generic;
using domain.Dto.crm;
using domain.Entities.common;
using domain.Entities.releastate;
using domain.Entities.users;

namespace domain.Entities.crm
{
    public class Company : BaseEntity
    {
        public Company()
        {
            Users = new HashSet<User>();
            Catalogs = new HashSet<Catalog>();
        }
        //Database Atributes
        //public int StatusId { get; set; }
        public int? CompanyTypeId { get; set; }
        public string? CompanyName { get; set; }
        public DateTime? RegistrationDate { get; set; }
        public string? Description { get; set; }
        public string? Site { get; set; }
        public string? GID { get; set; }
        public string? GVAT { get; set; }
        public string? WebSite { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? BillingAddress { get; set; }
        public string? ShipingAddress { get; set; }
        public string? BillingZip { get; set; }
        public string? ShipingZip { get; set; }
        public string? Fax { get; set; }
        public string? CompanyAcronym { get; set; }
        public string? ImagePath { get; set; }
        //Referent table
        //public Catalog Catalog { get; set; }
        public virtual CompanyType CompanyType { get; set; }
        //Table Sets
        public virtual ICollection<User>? Users { get; set; }
        public virtual ICollection<Catalog>? Catalogs { get; set; }

    }
}

File: GUI/seminar/src/app/models/Company.ts
import { User } from "./User";

export class Company {
  statusId: number = 0;
  companyTypeId: number = 0;
  companyName: string = "";
  description: string = "";
  phone: string = "";
  webSite: string = "";
  email: string = "";
  billingAddress: string = "";
  shipingAddress: string = "";
  billingCityId: number = 0;
  shipingCityId: number = 0;
  billingZip: string = "";
  shipingZip: string = "";
  leadId: number = 0;
  registrationDate: Date | null = null;
  gid: string = "";
  gvat: string = "";
  fax: string = "";
  billingCityStateId: number = 0;
  billingCityStateCountryId: number = 0;
  shippingCityStateId: number = 0;
  shippingCityStateCountryId: number = 0;
  companyTypeName: string = "";
  companyAcronym: string = "";
  id: number = 0;
  createdDate: Date | null = null;
  createdUser: string = "";
  modifiedDate: Date | null = null;
  modifiedUser: string = "";
  imagePath: string = "";
  imageFile?: File;

  users: User[] = [];

  constructor() {
    this.id = 0;
  }
}
